### ISSUE TITLE
latex(pdf) sets local tables of contents (or more generally topic nodes) in unbreakable boxes, causes overflow at bottom

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#2446]
When there are too many sections in a chapter, the

```
.. contents::
```

directive will create an unbreakable box, which does not allow page breaks. It is moved to next page and there overflows the bottom of the page. Example:

![capture d ecran 2016-04-13 a 17 27 25](https://cloud.githubusercontent.com/assets/2589111/14499069/07df867a-019d-11e6-9d9e-5a991a53cb5a.png)

Source file `index.rst` :

``` rst
==============
TEST SHADOWBOX
==============


TEST
====

.. contents::

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

Foo

---

Hello

```

The problem will apply to any `topic` node. Indeed the LaTeX Sphinx writer specifies to use various nested unbreakable boxes, the inner most once being `\shadowbox` from package `fancybox.sty`. 

Proposed fix: we should be able to  mimick the functionality using the already required package `framed.sty`. The "shadow box" effect is surely feasible using `framed`. And the box will then be splittable  on page breaks.

**Comment by jfbu** [comment]
But unfortunately package `framed.sty` does not allow (i.e. will lose) footnotes.

More modern LaTeX package `mdframed.sty` does handle footnotes but it exists only since 2010.

**Comment by jfbu** [comment]
@tk0miya 

I have a possibly working solution, using `framed.sty` which is already required. Maybe there is no problem with footnotes, because in fact they are suppressed inside `topic` node. This would replace entirely the `\shadowbox` from package `fancybox.sty` which _can not split across pages_.

But I have a problem of backwards perfect compatibility: in fact the current state is very buggy from the LaTeX point of view. Inside the LaTeX writer one finds:

``` python
    def visit_topic(self, node):
        self.in_minipage = 1
        self.body.append('\\setbox0\\vbox{\n'
                         '\\begin{minipage}{0.95\\linewidth}\n')

    def depart_topic(self, node):
        self.in_minipage = 0
        self.body.append('\\end{minipage}}\n'
                         '\\begin{center}\\setlength{\\fboxsep}{5pt}'
                         '\\shadowbox{\\box0}\\end{center}\n')
```

This generates LaTeX code which is **wrong**.
1. the `\vbox` works with the current line width. The minipage will have `95%` of that but the vbox itself will be `100%` of that. The `\vbox` is lacking a `\hsize0.95\linewidth`. As a result the minipage is padded with white space on the right.
2. that means that the framing encloses not only `95%` but `100%` of horizontal line. Then there is the extra `5pt` on each top, right, bottom, left. Thus we have `100% line width + 5pt + 5pt` as total horizontal space.
3. the center environment of LaTeX does _not_ know how to center things which are wider than the text width: it sets them left aligned, and thus the thing goes over to the right margin.

You can observe this on any example: the shaded box is wider than the line width, it is aligned at the left margin and spills over into the right margin.

Now, it is hard for me to maintain this. It is much easier for me to use `framed.sty` in a way such that the shadowed box fits _exactly_ in the line width: that means it goes exactly from left to right margin without spilling over.

Here is an image with current situation:

![capture d ecran 2016-04-15 a 20 25 35](https://cloud.githubusercontent.com/assets/2589111/14571286/b3f33356-0348-11e6-956d-c64ba796c656.png)

I added manually two horizontal rules to LaTeX file to display what happens. The inner rule shows that there is white space on the right, this is point 1. above. The outer rule below the box helps showing how in fact the box is _not_ centered but spills over in right margin.

Here is what I can do for example:

![capture d ecran 2016-04-15 a 20 28 22](https://cloud.githubusercontent.com/assets/2589111/14571331/e57c55ec-0348-11e6-9e26-707249ce8534.png)

Here the shadow box fits perfectly in the textwidth. There is some lost vertical space below (the currently used `center` adds it), but I can fix that to give same result as before.

What should I do ? sticking to the old looks is annoying, and it complicates things as  `framed.sty` by design will try to center the whole thing (it has no problem spilling over in margins, but symmetrically; in the example above, I used its ability to fit the whole exactly in the line width).

**Comment by jfbu** [comment]
Finally I decided for the PR #2453 that it was more aesthetic with the "shadow" in the margin, and the frame precisely at the margin (on both sides). It was difficult to obtain exact same vertical spacing above and below, because the former situation was the result of buggy LaTeX (for example a too wide line can not be centered with `center` environment and even it creates an extra white line above; I tried to mimic that). The differences in vertical spaces are very slight.

For very long such topic boxes which will now split, one could think about distinct designs for the first, medium and last parts, but I simply replicate the "shadow box" for all.
